vericoding/core/config.py

```diff
@@ -0,0 +1,186 @@
+"""Configuration management for vericoding."""
+
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any
+
+try:
+    import tomllib  # Python 3.11+
+except ImportError:
+    try:
+        import tomli as tomllib  # Fallback for older Python versions
+    except ImportError as e:
+        raise ImportError(
+            "TOML support requires Python 3.11+ or the 'tomli' package. "
+            "Install with: pip install tomli"
+        ) from e
+
+
+# Load environment variables from .env file
+def load_environment():
+    """Load environment variables from .env file if it exists."""
+    try:
+        from dotenv import load_dotenv
+
+        # Look for .env file in current directory and parent directories
+        env_file = Path(".env")
+        if not env_file.exists():
+            # Try parent directory (useful when running from subdirectories)
+            env_file = Path("../.env")
+
+        if env_file.exists():
+            load_dotenv(env_file)
+            print(f"✓ Loaded environment variables from {env_file}")
+        else:
+            # Try to load from default location
+            load_dotenv()
+    except ImportError:
+        # Fallback if dotenv is not installed
+        print(
+            "Note: python-dotenv not installed. Using system environment variables only."
```

Comment:
The fallback shouldn't be there since we control the env and use uv. And dotenv is guaranteed